src/app/zap-templates/templates/chip/CHIPClusters-src.zapt

```diff
@@ -2,18 +2,32 @@
 #include "CHIPClusters.h"
 
 #include <app/chip-zcl-zpro-codec-api.h>
+#include <app/im-encoder.h>
+#include <support/ReturnMacros.h>
 
 namespace chip {
 namespace Controller {
 
+// TODO(@vivien-apple): onCompletion is not used by IM for now.
+// TODO: length should be passed to commands when byte string is in argument list.
+// TODO: Commands should take group id as an argument.
+
 {{#chip_server_clusters}}
 
 // {{asCamelCased name false}} Cluster Commands
 {{#chip_server_cluster_commands}}
 CHIP_ERROR {{asCamelCased clusterName false}}Cluster::{{asCamelCased name false}}(Callback::Callback<> * onCompletion{{#chip_server_cluster_command_arguments}}, {{chipType}} {{asCamelCased label}}{{/chip_server_cluster_command_arguments}})
 {
+#ifdef CHIP_APP_USE_INTERACTION_MODEL
+    VerifyOrReturnError(mDevice != nullptr, CHIP_ERROR_INCORRECT_STATE);
+    (void) onCompletion;
+    // TODO(@vivien-apple): onCompletion is not used by IM for now.
+    chip::app::cluster::{{asCamelCased clusterName false}}::Encode{{asType name}}Command(mDevice->GetCommandSender(), mEndpoint, 0{{#chip_server_cluster_command_arguments}}, {{asCamelCased label}}{{/chip_server_cluster_command_arguments}});
```

Comment:
Again I would like the command encoding code to be part of `CHIPClusters-src.zapt`.

Trying to get rid of `encoder.cpp` I already have to deal with code that depends on the API exposed by it and adding a new dependency on a new API would just make migration harder.

If we need device to generates command (and we will) we need to find a proper way to get them to access the `CHIPCluster` API. IMHO it would be easier for us to have a single `cpp` API that is used both for the accessory and for the controller.